Source/Core/Core/AchievementManager.cpp

```diff
@@ -162,6 +164,26 @@ void AchievementManager::FetchPlayerBadge()
 
 void AchievementManager::FetchGameBadges()
 {
+  rc_client_achievement_list_t* achievement_list;
+  {
+    std::lock_guard lg{m_lock};
+    achievement_list = rc_client_create_achievement_list(
+        m_client, RC_CLIENT_ACHIEVEMENT_CATEGORY_CORE_AND_UNOFFICIAL,
+        RC_CLIENT_ACHIEVEMENT_LIST_GROUPING_PROGRESS);
+  }
+  for (u32 bx = 0; bx < achievement_list->num_buckets; bx++)
+  {
+    auto& bucket = achievement_list->buckets[bx];
+    for (u32 achievement = 0; achievement < bucket.num_achievements; achievement++)
+    {
+      u32 achievement_id = bucket.achievements[achievement]->id;
+      // I need these defaults to be set before making any FetchBadge calls, because FetchBadge
+      // will update the UI and attempt to retrieve these badges.
+      m_unlocked_badges[achievement_id] = m_default_unlocked_badge;
+      m_locked_badges[achievement_id] = m_default_locked_badge;
+    }
+  }
```

Comment:
This seems rather wasteful of memory, and also unnecessary. You already have a `GetAchievementBadge()` function that falls back to the default icon if it's not in the `m_unlocked_badges`/`m_locked_badges` maps, just use that consistently everywhere?